ext/testlib/loader.py

```diff
@@ -221,22 +222,28 @@ def load_file(self, path):
         sys.path.insert(0, os.path.dirname(path))
         cwd = os.getcwd()
         os.chdir(os.path.dirname(path))
-        config.config.file_under_load = path
+        configuration.config.file_under_load = path
 
         new_tests = test_mod.TestCase.collector.create()
         new_suites = suite_mod.TestSuite.collector.create()
         new_fixtures = fixture_mod.Fixture.collector.create()
 
+        print(test_mod)
+        print(test_mod.TestCase)
+        print(test_mod.TestCase.collector)
+
         def cleanup():
-            config.config.file_under_load = None
+            configuration.config.file_under_load = None
             sys.path[:] = old_path
             os.chdir(cwd)
             test_mod.TestCase.collector.remove(new_tests)
             suite_mod.TestSuite.collector.remove(new_suites)
             fixture_mod.Fixture.collector.remove(new_fixtures)
 
         try:
-            execfile(path, newdict, newdict)
+            with open(path, "rb") as f:
+                test = compile(f.read(), path, "exec")
+            exec(test, newdict, newdict)
```

Comment:
Why not just
```suggestion
            exec(open(path).read(), newdict, newdict)
```
https://stackoverflow.com/questions/436198/what-is-an-alternative-to-execfile-in-python-3

Reply:
That `open` statement is not guaranteed to be closed if it fails, but I'm not sure whether it is necessary.

Reply:
That `open` statement is not guaranteed to be closed if it fails, but I'm not sure whether it is necessary.